### ISSUE TITLE
deprecate graduated nd filter - only produces artifacts/halos

### ISSUE THREAD

**Issue by blitzgneisserin** [darktable-org/darktable#13840]
Hi,

I would like to suggest to add the graduated filter module to the deprecated modules. It only produces artifacts and halos. People can use exposure with a mask and colorize instead.

I think I know how to do this in the code, although I have not tried yet.

I can add examples later if people don't know what I mean.

Unfortunately I don't know how to fix this in the module and I suspect it's not so easy to fix since this "bug" has been present for many years now.

**Comment by TurboGit** [comment]
> I can add examples later if people don't know what I mean.

Yes please. I do use this module from time to time and it works fine for me. Maybe there is something we can fix instead of deprecating it. But we need some more data.

**Comment by blitzgneisserin** [comment]
I mean this stripe shaped artifact in particular
![2023-03-06-153957_1920x1200_scrot](https://user-images.githubusercontent.com/16564210/223142564-f33d01b9-4e5c-4802-9dc9-2401694431e6.png)
I think this is some wrong calculation, but I don't know enough math to fix it

**Comment by TurboGit** [comment]
> I think this is some wrong calculation, but I don't know enough math to fix it

Probably so yes. But you are also using a very strong effect +4.7EV. Another point (as you said) is that two modules are needed for the same effect so deprecating is not a good option.

**Comment by jenshannoschwalm** [comment]
I think the reported banding is a known problem of the module (mentioned in the docs also). I usually tackle this by a smaller EV and a sharper effect transition.
From reading the code, a fix would more be a rewrite ...

**Comment by HansBull** [comment]
Exposure with a gradient mask could be a solution. At least that's what I am using.

**Comment by HansBull** [comment]
Exposure with a gradient mask could be a solution. At least that's what I am using.

**Comment by TurboGit** [comment]
> Exposure with a gradient mask could be a solution. At least that's what I am using.

But is missing the possibility to "color" the gradient. You'll need to masks in two modules to achieve that or maybe a single using the "color balance rgb", but that's a quite heavy module GUI wise for a simple gradient :)

**Comment by blitzgneisserin** [comment]
It is easy to use/create one or more styles with exposure+mask+color balance instead
Come on guys, a broken tool is not good for the image, I mean the image of dt. So I am still in favour of deprecating

**Comment by blitzgneisserin** [comment]
Btw, thanks to everybody who is participating in this discussion

**Comment by victoryforce** [comment]
> It is easy to use/create one or more styles with exposure+mask+color balance instead

Everything is simple if you are familiar with all the tools of the program. But there are things that will not be easy for you if you are a beginner.

> Come on guys, a broken tool is not good for the image, I mean the image of dt. So I am still in favour of deprecating

- There is nothing inherently unfixable in this module.
- A tool that works imperfectly is not a broken tool. This is a tool that can be improved.

**Comment by TurboGit** [comment]
I fully agree with you @victoryforce !
